replica/table.cc

```diff
@@ -2305,7 +2305,7 @@ future<table::snapshot_file_set> table::take_snapshot(database& db, sstring json
     auto table_names = std::make_unique<std::unordered_set<sstring>>();
 
     co_await io_check([&jsondir] { return recursive_touch_directory(jsondir); });
-    co_await db.get_sharded_sst_dir_semaphore().local().parallel_for_each(tables, [&jsondir, &table_names] (sstables::shared_sstable sstable) {
+    co_await _sstables_manager.dir_semaphore().parallel_for_each(tables, [&jsondir, &table_names] (sstables::shared_sstable sstable) {
```

Comment:
Is `db` param even used after this? I think it can be removed.

Reply:
Indeed. I remember that I checked `snapshot_on_all_shards()` for the need in db, and it still was required :( But it can be removed from `take_snapshot()`